raft/node.go

```diff
@@ -144,7 +152,8 @@ type Node interface {
 	Step(ctx context.Context, msg pb.Message) error
 
 	// Ready returns a channel that returns the current point-in-time state.
-	// Users of the Node must call Advance after retrieving the state returned by Ready.
+	// Users of the Node must call Advance after retrieving the state returned by Ready (unless
+	// async storage writes is enabled, in which case it should never be called).
```

Comment:
I hear what you're saying, but IMO that leaves room for bugs. I'll keep this as a panic to avoid misuse. It's easy enough to only call `Advance` when `!AsyncStorageWritesEnabled` as a user of this library.